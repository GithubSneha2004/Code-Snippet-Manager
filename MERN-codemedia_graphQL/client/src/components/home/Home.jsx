import React from 'react';
import { Link } from 'react-router-dom';
import homeImage from '../../assets/homeimage-removebg-preview.png';
import './home.css';

export default function Home() {
  return (
    <section className="bg-white dark:bg-gray-900 py-16">
      <div className="container mx-auto flex flex-col-reverse lg:flex-row items-center justify-between px-6">
        
        {/* Left Side - Text */}
        <div className="w-full lg:w-1/2 mt-10 lg:mt-0">
          <h1 className="text-4xl font-extrabold text-gray-900 dark:text-white mb-4 text-center lg:text-left">
            <span className="text-blue-600">Code Snippet Manager</span>
          </h1>
          <p className="mb-4 text-gray-700 dark:text-gray-300 text-lg text-justify">
            Welcome to <strong>Code Snippet Manager</strong>, the ultimate tool for organizing and storing your code snippets. Whether you're a developer, designer, or hobbyist, our app will help you keep your code snippets organized and easily accessible.
          </p>
          <p className="mb-6 text-gray-700 dark:text-gray-300 text-lg text-justify">
            Start managing your code snippets today and experience the benefits of an efficient workflow. Sign up or log in to get started!
          </p>
          <div className="text-center lg:text-left">
            <Link
              to="/signup"
              className="inline-flex items-center justify-center px-6 py-3 text-white bg-blue-600 hover:bg-blue-700 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-lg transition duration-300"
            >
              Get Started
            </Link>
          </div>
        </div>

        {/* Right Side - Image */}
        <div className="w-full lg:w-1/2 flex justify-center">
          <img
            src={homeImage}
            alt="Code Snippet Illustration"
            className="max-w-full h-auto rounded-xl shadow-lg"
          />
        </div>
      </div>
    </section>
  );
}
