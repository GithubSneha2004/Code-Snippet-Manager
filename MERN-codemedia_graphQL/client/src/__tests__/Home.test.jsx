import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from '../components/home/Home';

describe('Home Component', () => {
  beforeEach(() => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );
  });

  test('renders main heading', () => {
    const heading = screen.getByRole('heading', { name: /Code Snippet Manager/i });
    expect(heading).toBeInTheDocument();
  });

  test('renders descriptive paragraphs', () => {
    expect(screen.getByText(/ultimate tool for organizing and storing your code snippets/i)).toBeInTheDocument();
    expect(screen.getByText(/start managing your code snippets today/i)).toBeInTheDocument();
  });

  test('renders Get Started link with correct href', () => {
    const getStartedLink = screen.getByRole('link', { name: /Get Started/i });
    expect(getStartedLink).toBeInTheDocument();
    expect(getStartedLink).toHaveAttribute('href', '/signup');
  });

  test('renders image with alt text', () => {
    const img = screen.getByAltText(/Code Snippet Illustration/i);
    expect(img).toBeInTheDocument();
  });
});
