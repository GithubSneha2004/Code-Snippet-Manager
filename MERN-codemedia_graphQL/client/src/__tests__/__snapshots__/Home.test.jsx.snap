// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches snapshot 1`] = `
<div>
  <section
    class="bg-white dark:bg-gray-900 py-16"
    data-testid="home-section"
  >
    <div
      class="container mx-auto flex flex-col-reverse lg:flex-row items-center justify-between px-6"
    >
      <div
        class="w-full lg:w-1/2 mt-10 lg:mt-0"
      >
        <h1
          class="text-4xl font-extrabold text-gray-900 dark:text-white mb-4 text-center lg:text-left"
        >
          <span
            class="text-blue-600"
          >
            Code Snippet Manager
          </span>
        </h1>
        <p
          class="mb-4 text-gray-700 dark:text-gray-300 text-lg text-justify"
        >
          Welcome to 
          <strong>
            Code Snippet Manager
          </strong>
          , the ultimate tool for organizing and storing your code snippets. Whether you're a developer, designer, or hobbyist, our app will help you keep your code snippets organized and easily accessible.
        </p>
        <p
          class="mb-6 text-gray-700 dark:text-gray-300 text-lg text-justify"
        >
          Start managing your code snippets today and experience the benefits of an efficient workflow. Sign up or log in to get started!
        </p>
        <div
          class="text-center lg:text-left"
        >
          <a
            class="inline-flex items-center justify-center px-6 py-3 text-white bg-blue-600 hover:bg-blue-700 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-lg transition duration-300"
            href="/signup"
          >
            Get Started
          </a>
        </div>
      </div>
      <div
        class="w-full lg:w-1/2 flex justify-center"
      >
        <img
          alt="Code Snippet Illustration"
          class="max-w-full h-auto rounded-xl shadow-lg"
          src="homeimage-removebg-preview.png"
        />
      </div>
    </div>
  </section>
</div>
`;
